domain/ide/workspace/ui/diagnostics_overview.go

```diff
@@ -0,0 +1,204 @@
+			return
+		}
+
+		diagnosticsOverviewParams := types.DiagnosticsOverviewParams{Product: p.ToProductCodename(), Html: htmlBuffer.String()}
+		notifier.Send(diagnosticsOverviewParams)
+		logger.Debug().Msgf("sent diagnostics overview htmlBuffer for product %s", p)
+		logger.Trace().
+			Int("issueCount", len(issuesByFile)).
+			Any("diagnosticsOverviewParams", diagnosticsOverviewParams).
+			Msg("detailed tree data")
+	}
+}
+
+func getFileNodes(issuesByFile snyk.IssuesByFile) map[Node][]Node {
+	fileNodes := make(map[Node][]Node)
+	for path, issues := range issuesByFile {
+		fileNode := Node{
+			Icon: getFileTypeIcon(),
+			Text: template.HTML(path),
+		}
+		issueNodes := []Node{}
+		for _, issue := range issues {
+			issueNodes = append(issueNodes, Node{
+				Icon: html.GetSeverityIconSvg(issue),
+				Text: template.HTML(issue.AdditionalData.GetTitle()),
+			})
+		}
+		fileNodes[fileNode] = issueNodes
+	}
+	return fileNodes
+}
+
+func getFileTypeIcon() template.HTML {
+	return ""
+}
+
+// generateSecurityNonce generates a cryptographically secure random nonce.
+// A nonce is used in the Content Security Policy (CSP) to allow specific
+// inline styles and scripts, helping to prevent Cross-Site Scripting (XSS) attacks.
+func generateSecurityNonce() (string, error) {
+	nonceBytes := make([]byte, 16)
+	if _, err := rand.Read(nonceBytes); err != nil {
+		return "", fmt.Errorf("error generating nonce: %v", err)
+	}
+	return base64.StdEncoding.EncodeToString(nonceBytes), nil
+}
+
+func getRootNodes(c *config.Config, p product.Product, issuesByFile snyk.IssuesByFile) []Node {
+	var icon template.HTML
+
+	productEnabled := isProductEnabled(c, p)
+	if productEnabled {
+		icon = html.GetProductIcon(p)
+	} else {
+		icon = html.GetProductIconDisabled(p)
+	}
+
+	rootNodeTitle := getRootNodeText(issuesByFile, p)
+
+	rootNodes := append([]Node{}, Node{
+		Icon:           icon,
+		Text:           template.HTML(rootNodeTitle),
+		ProductEnabled: productEnabled,
+	})
+
+	fixableCount := issuesByFile.FixableCount()
+	if fixableCount > 0 {
+		plural := ""
+		if fixableCount > 1 {
+			plural = "s"
+		}
+		rootNodes = append(rootNodes, Node{
+			Text: template.HTML(fmt.Sprintf("⚡️ %d issue%s can be fixed automatically", fixableCount, plural)),
+		})
+	}
+
+	return rootNodes
+}
+
+func getRootNodeText(issuesByFile snyk.IssuesByFile, p product.Product) string {
+	total, critical, high, medium, low := issuesByFile.SeverityCounts()
+
+	pluralSuffix := ""
+	if total > 1 {
+		pluralSuffix = "s"
+	}
+
+	var rootNodeTitle = fmt.Sprintf("%s - No issues found", p.ToFilterableIssueType()[0])
+	if total > 0 {
+		rootNodeTitle = fmt.Sprintf(
+			"%s - %d unique issue%s: %s",
+			p.ToFilterableIssueType()[0],
+			total,
+			pluralSuffix,
+			issuesByFile.SeverityCountsAsString(critical, high, medium, low),
+		)
+	}
+	return rootNodeTitle
+}
+
+func isProductEnabled(c *config.Config, p product.Product) bool {
+	switch p {
+	case product.ProductOpenSource:
```

Comment:
should we move this to config?

Reply:
Yeah, I was originally thinking of having it in product, but then it would have cyclical dependencies. But `config` sounds good.

Reply:
Actually, we have that code twice - just realized it's already in scanNotifier - so one more reason to move it.